indico/testing/fixtures/database_docker.py

```diff
@@ -0,0 +1,207 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2025 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+import os
+import shutil
+import tempfile
+import time
+import warnings
+from collections.abc import Iterator
+from contextlib import contextmanager
+from urllib.parse import urlparse
+
+import docker
+from docker.errors import NotFound
+from docker.models.containers import Container
+
+
+READY_TIMEOUT_SEC = 20
+CONTAINER_NAME_TPL = 'test-indico-pg-{pid}'
+POSTGRES_IMAGE = 'docker.io/postgres:16'
+DB_NAME = 'test'
+
+
+def create_container_low_level(
+    api_client: docker.APIClient, container_name: str, socket_dir: str, data_dir: str
+) -> str:
+    """
+    Create a PostgreSQL container using the low-level docker-py API.
+
+    :param api_client: The low-level Docker API client.
+    :param container_name: The name to assign to the new container.
+    :param socket_dir: Path to the host directory to mount as the PostgreSQL socket directory.
+    :param data_dir: Path to the host directory to mount as the PostgreSQL data directory.
+
+    :return: The ID of the created Docker container.
+    :raises docker.errors.APIError: If the container creation fails.
+    """
+    # TODO: once https://github.com/docker/docker-py/issues/3351 is solved, use the high level API
+
+    host_config = api_client.create_host_config(
+        binds={
+            socket_dir: {'bind': '/var/run/postgresql', 'mode': 'rw'},
+            data_dir: {'bind': '/var/lib/postgresql/data', 'mode': 'rw'},
```

Comment:
Does that expose the host-level `/var/lib/postgresql/data`? If yes this seems to be quite dangerous if you run a local Postgres as well...